pkg/playback/http_playback.go

```diff
@@ -0,0 +1,413 @@
+			rr.httpRecorder.webhookHandler(w, r)
+		} else {
+			fmt.Println("Error: webhook endpoint should never be called in replay mode")
+		}
+	}
+}
+
+// note: calling this method always sets cassetteLoaded = false
+func (rr *Server) switchMode(modeString string) error {
+	rr.cassetteLoaded = false
+	switch strings.ToLower(modeString) {
+	case Record:
+		rr.mode = Record
+		return nil
+	case Replay:
+		rr.mode = Replay
+		return nil
+	case Auto:
+		rr.mode = Auto
+		return nil
+	default:
+		return fmt.Errorf("\"%s\" is not a valid playback mode. It must be either \"record\", \"replay\", or \"auto\"", modeString)
+	}
+}
+
+func (rr *Server) setCassetteDir(absoluteCassetteDir string) error {
+	handle, err := os.Stat(absoluteCassetteDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return fmt.Errorf("the directory \"%v\" does not exist. Please create it, then try again", absoluteCassetteDir)
+		}
+		return fmt.Errorf("inexpected error when checking cassette directory: %w", err)
+	}
+
+	if !handle.Mode().IsDir() {
+		return fmt.Errorf("the path \"%v\" is not a directory", absoluteCassetteDir)
+	}
+
+	rr.cassetteDirectory = absoluteCassetteDir
+	return nil
+}
+
+// loadCassette() takes a relative path (relative to the rr.cassetteDirectory) to a .yaml file
+// The .yaml file need not exist (unless rr.mode = Replay). If intermediate directories in the relative path do not exist, loadCassette *will* create them.
+func (rr *Server) loadCassette(relativeFilepath string) error {
+	absoluteFilepath := filepath.Join(rr.cassetteDirectory, relativeFilepath)
+
+	var shouldCreateNewFile bool
+
+	switch rr.mode {
+	case Record:
+		shouldCreateNewFile = true
+	case Replay:
+		shouldCreateNewFile = false
+	case Auto:
+		_, err := os.Stat(absoluteFilepath)
+		if os.IsNotExist(err) {
+			shouldCreateNewFile = true
+			rr.isRecordingInAutoMode = true
+		} else {
+			shouldCreateNewFile = false
+			rr.isRecordingInAutoMode = false
+		}
+	}
+
+	if shouldCreateNewFile {
+		directoryPath := filepath.Dir(absoluteFilepath)
+		err := os.MkdirAll(directoryPath, 0755)
+		if err != nil {
+			return fmt.Errorf("Error recursively creating nested directories for cassette file: %w", err)
+		}
+
+		fileHandle, err := os.Create(absoluteFilepath)
+		if err != nil {
+			return fmt.Errorf("Error creating cassette file: %w", err)
+		}
+
+		err = rr.httpRecorder.insertCassette(fileHandle)
+		if err != nil {
+			return fmt.Errorf("Error inserting cassette file: %w", err)
+		}
+	} else {
+		fileHandle, err := os.Open(absoluteFilepath)
+		if err != nil {
+			return fmt.Errorf("Error opening cassette file: %w", err)
+		}
+
+		err = rr.httpReplayer.readCassette(fileHandle)
+		if err != nil {
+			return fmt.Errorf("Error parsing cassette file: %v", err)
+		}
+	}
+
+	rr.cassetteLoaded = true
+	return nil
+}
+
+func (rr *Server) ejectCassette() error {
+	if !rr.cassetteLoaded {
+		return fmt.Errorf("tried to eject when no cassette is loaded")
+	}
+
+	switch rr.mode {
```

Comment:
(this switch also can be deduped if you implement `rr.isRecording()` )

Reply:
Done in 961d13ecaf1f0c8830e80fe070101dbe0ce126ab